helmfile.d/charts/prometheus-alerts/files/webhooks.yaml

```diff
@@ -0,0 +1,10 @@
+- name: webhook-failures
+  rules:
+  - alert: WebhookFailing
+    annotations:
+      description: Webhooks have been failing during the last 10m
+      summary: Webhooks requests are failing
+    expr: rate(apiserver_admission_webhook_rejection_count{error_type!="no_error"}[5m]) > 0
```

Comment:
Does the count for `apiserver_admission_webhook_rejection_count` increase if, let's say, gatekeper restarts and one of the pods happens to handle a request (not sure if that's a valid scenario or not)?

Reply:
I believe this should increase the counter (simply scaling down to 0 replicas of the gatekeeper controller seems to always increase the counter). This alone is not enough to trigger the alert from my testing, as it shouldn't keep the `rate` of the counter above 0 for a long enough period alone.

The following screenshot is from my testing after scaling down to 0 replicas and just trying to create a deployment continuously over a 10 minute period, which fails because gatekeeper is unavailable. This triggers the alert.

The second spike is me just trying to create a deployment a single time, then scaling up to 3 replicas again, which does not trigger the alert.

From my testing, the `error_type!="no_error"` (could also be `error_type="calling_webhook_error"`) should filter out every request that gets rejected due to violating a policy, i.e. trying to create a deployment using the latest tag, and only take into account request that actually fail.

![Screenshot from 2024-01-12 07-33-21](https://github.com/elastisys/compliantkubernetes-apps/assets/60571221/f5347624-3570-4a82-b402-736740f2b9a1)

Reply:
@OlleLarsson Any thoughts on this? We could lower the interval for the alert to trigger but my fear is that it might get too spammy

Reply:
We can adjust interval if needed further down the road.